Makefile.in

```
@@ -722,41 +722,41 @@ distclean: clean
 ################################################################################
 
 install: library | $(INSTALL_DIRS)
-	$(CP) flint.pc $(PKGCONFIGDIR)/flint.pc
+	$(CP) flint.pc $(DESTDIR)$(PKGCONFIGDIR)/flint.pc
 ifneq ($(SHARED), 0)
 ifneq ($(FLINT_DLLLIB),0)
-	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB) $(BINDIR)
-	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_MAJOR) $(BINDIR)
-	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_FULL) $(BINDIR)
-	$(CP) $(FLINT_DIR)/$(FLINT_IMPLIB) $(LIBDIR)
+	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB) $(DESTDIR)$(BINDIR)
+	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_MAJOR) $(DESTDIR)$(BINDIR)
+	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_FULL) $(DESTDIR)$(BINDIR)
+	$(CP) $(FLINT_DIR)/$(FLINT_IMPLIB) $(DESTDIR)$(LIBDIR)
 else
-	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB) $(LIBDIR)
-	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_MAJOR) $(LIBDIR)
-	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_FULL) $(LIBDIR)
+	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB) $(DESTDIR)$(LIBDIR)
+	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_MAJOR) $(DESTDIR)$(LIBDIR)
+	$(CP_A) $(FLINT_DIR)/$(FLINT_LIB_FULL) $(DESTDIR)$(LIBDIR)
 endif
 ifneq ($(FLINT_DYLIB),0)
-	install_name_tool -id $(LIBDIR)/$(FLINT_LIB_FULL) $(LIBDIR)/$(FLINT_LIB)
+	install_name_tool -id $(LIBDIR)/$(FLINT_LIB_FULL) $(DESTDIR)$(LIBDIR)/$(FLINT_LIB)
```

```suggestion
	install_name_tool -id @libdir@/$(FLINT_LIB_FULL) $(LIBDIR)/$(FLINT_LIB)
```
What about just fixing this and leaving all the rest unchanged?

I worry that in the future some added line might forget to include `$(DESTDIR)`.

Alternatively something like this:
```suggestion
	install_name_tool -id $(FLINT_DYLIB_ID) $(LIBDIR)/$(FLINT_LIB)
```
where this is added at the top:
```
FLINT_DYLIB_ID:=@libdir@/@FLINT_LIB_FULL@
```

---

I did it that way because I thought it more standard for things like `$(BINDIR)` to refer to final paths, with $(DESTDIR) only coming up in the installation target. But I'm happy with any of these options.

---

I don't know, maybe you are right. Don't take my suggestion as a reference you probably know better than me.

In any case, I think `$(DESTDIR)` handling has been broken a few times in flint in the past, so whatever minimizes chances of getting it wrong is good.